daemon/daemon.go

```diff
@@ -1476,15 +1476,15 @@ func (daemon *Daemon) IdentityMapping() idtools.IdentityMapping {
 }
 
 // ImageService returns the Daemon's ImageService
-func (daemon *Daemon) ImageService() *images.ImageService {
+func (daemon *Daemon) ImageService() ImageService {
```

Comment:
This is where the services is initiated; I think that should usually be a concrete type, not an interface; as long as it satisfies the interface for the receiver, do we need to change it? (possibly we don't)

Reply:
> This is where the services is initiated;

Actually no, the service is initiated in [NewDaemon](https://github.com/moby/moby/blob/master/daemon/daemon.go#L1086), this method publishes that service. 

In an ideal world the one creating the daemon would create the image service and pass it to `NewDaemon` and the daemon wouldn't publish that field.

Reply:
ImageService should be passed to daemon by NewDaemon and kept encapsulated then. I haven't checked where this func is used, but I would expect this is not necessary, or other components just need their lifecycle to be fixed so they don't need to lookup for dependent components like this.

Reply:
Yes, we were talking about this with @thaJeztah, at one point we should clean up the startup code and make it so that things don't create their own dependencies.

Reply:
who said "dependency injection" :P